src/ctia/init.clj

```diff
@@ -41,80 +40,76 @@
    ;; result.  This is where you can do idempotent store setup.
    ;; Specifying a :builder is optional.
    :builder
-   {:memory (fn memory-builder [factory]
-              (factory (atom {})))
-    :es (fn es-builder [factory]
-          (when (es-store/uninitialized?)
-            (es-store/init!))
-          (factory @es-store/es-state))
-    :sql (fn sql-builder [factory]
-           (when (sql-store/uninitialized?)
-             (sql-store/init!))
+   {:atom (fn atom-builder [factory props]
+            (factory (atom {})))
+    :es (fn es-builder [factory props]
+          (factory (es-store/init! props)))
+    :sql (fn sql-builder [factory props]
+           (sql-store/init! props)
            (factory))}
 
    :actor
-   {:memory as/->ActorStore
+   {:atom as/->ActorStore
     :es es-store/->ActorStore}
 
    :campaign
-   {:memory as/->CampaignStore
+   {:atom as/->CampaignStore
     :es es-store/->CampaignStore}
 
    :coa
-   {:memory as/->COAStore
+   {:atom as/->COAStore
     :es es-store/->COAStore}
 
    :exploit-target
-   {:memory as/->ExploitTargetStore
+   {:atom as/->ExploitTargetStore
     :es es-store/->ExploitTargetStore}
 
    :feedback
-   {:memory as/->FeedbackStore
+   {:atom as/->FeedbackStore
     :es es-store/->FeedbackStore}
 
    :identity
-   {:memory as/->IdentityStore
+   {:atom as/->IdentityStore
     :es es-store/->IdentityStore}
 
    :incident
-   {:memory as/->IncidentStore
+   {:atom as/->IncidentStore
     :es es-store/->IncidentStore}
 
    :indicator
-   {:memory as/->IndicatorStore
+   {:atom as/->IndicatorStore
     :es es-store/->IndicatorStore}
 
    :judgement
-   {:memory as/->JudgementStore
+   {:atom as/->JudgementStore
     :es es-store/->JudgementStore
     :sql #(do (sql-judgement/init!)
               (ss/->JudgementStore))}
 
    :sighting
-   {:memory as/->SightingStore
+   {:atom as/->SightingStore
     :es es-store/->SightingStore}
 
    :ttp
-   {:memory as/->TTPStore
+   {:atom as/->TTPStore
     :es es-store/->TTPStore}})
 
 (defn init-store-service! []
   (store-factory-cleaner)
-  (doseq [[store-type store-atom] store/stores]
-    (let [selected-store (get-in @p/properties
-                                 [:ctia :store store-type])
-          builder (get-in store-factories
-                          [:builder selected-store]
-                          (fn default-builder [f] (f)))
+  (doseq [[store-name store-atom] store/stores]
+    (let [store-type (keyword (get-in @p/properties [:ctia :store store-name] "none"))
+          store-properties (merge (get-in @p/properties [:ctia :store store-type :default] {})
+                                  (get-in @p/properties [:ctia :store store-type store-name] {}))
```

Comment:
@polygloton will your `properties` atom be updatable at runtime?

If so, then this kind of code where you pull different fields off in succession is open to possible race conditions, where the properties are not consistent. It's a small edge case, but thought I'd point it out regardless.